src/witness_generator/generator.rs

```diff
@@ -305,13 +305,21 @@ where
         // Note that we should always query all machines that match, because they might
         // update their internal data, even if all values are already known.
         // TODO could it be that multiple machines match?
+        let mut fixed_lookup = None;
         for m in &mut self.machines {
             // TODO also consider the reasons above.
-            if let Some(result) =
-                m.process_plookup(self.fixed_data, identity.kind, &left, &identity.right)
-            {
+            if let Some(result) = m.process_plookup(
+                self.fixed_data,
+                &mut fixed_lookup,
+                identity.kind,
+                &left,
+                &identity.right,
+            ) {
                 return result;
```

Comment:
Doesn't returning early here contradict the comment line 305?

Reply:
Yes, it depends on the comment `TODO could it be that multiple machines match?` ;)